Src/VimCore/Interpreter_Interpreter.fs

```diff
@@ -807,6 +808,97 @@ type VimInterpreter
             if lineRange.Count > 1 then
                 _foldManager.CreateFold lineRange)
 
+    member x.RunNormal (lineRange: LineRangeSpecifier) input =
+        let transactionMap = System.Collections.Generic.Dictionary<ITextBuffer, ILinkedUndoTransaction>();
+        let modeSwitchMap = System.Collections.Generic.Dictionary<ITextBuffer, IVimBuffer>();

```

Comment:
So, why not use `IVimBuffer`? From your description, it seems to me like a better choice over `ITextBuffer` and even `ITextView`.

One more thing to consider: this map strategy I have copied from RunMacro method (on CommandUtil). It seems that we might need to change that implementation as well.